cmd/shortener/main.go

```diff
@@ -157,33 +242,89 @@ func main() {
 		fmt.Fprint(os.Stderr, err.Error())
 	}
 
-	defer util.GetLogger().Sync()
+	defer func() {
+		err = util.GetLogger().Sync()
+		if err != nil {
+			return
+		}
+	}()
 
 	util.GetLogger().Infoln("dsn", conf.DSN)
 
 	state.InitShortAddress(conf.ShortAddr.Addr)
 
-	r := router(conf.JSONFile, pg)
+	var wg sync.WaitGroup
+
+	r := router(conf.JSONFile, pg, &wg)
+
+	var m *autocert.Manager
+
+	if *httpsRequired != "" {
+		m = &autocert.Manager{
+			Cache:  autocert.DirCache(".cache"),
+			Prompt: autocert.AcceptTOS,
+		}
+
+		go func() {
+			h := m.HTTPHandler(nil)
+			fmt.Println(http.ListenAndServe(":80", h))
+		}()
+	}
 
 	util.GetLogger().Infoln(conf)
+
 	addrToServe := strings.TrimPrefix(conf.HTTPAddr.Addr, "http://")
+	addrToServe = strings.TrimPrefix(addrToServe, "https://")
```

Comment:
It would be better if you use `regex`

Reply:
https://stackoverflow.com/questions/24378324/remove-http-and-https-from-a-string